src/compute/src/compute_state.rs

```diff
@@ -259,16 +260,19 @@ impl<'a, A: Allocate + 'static> ActiveComputeState<'a, A> {
         match enable_columnation_lgalloc {
             Some(true) => {
                 if let Some(path) = self.compute_state.context.scratch_directory.as_ref() {
-                    info!(path = ?path, "Enabling lgalloc");
-                    lgalloc::lgalloc_set_config(
-                        lgalloc::LgAlloc::new()
-                            .enable()
-                            .with_path(path.clone())
-                            .with_background_config(lgalloc::BackgroundWorkerConfig {
-                                interval: Duration::from_secs(1),
-                                batch: 32,
-                            }),
-                    );
+                    info!(path = ?path, eager_return=enable_lgalloc_eager_reclamation, "Enabling lgalloc");
+                    let mut config = lgalloc::LgAlloc::new();
+                    config
+                        .enable()
+                        .with_path(path.clone())
+                        .with_background_config(lgalloc::BackgroundWorkerConfig {
+                            interval: Duration::from_secs(1),
+                            batch: 32,
+                        });
+                    if let Some(eager_return) = enable_lgalloc_eager_reclamation {
+                        config.eager_return(eager_return);
+                    }
+                    lgalloc::lgalloc_set_config(&config);
```

Comment:
This works, but only because adapter always sends all config options in every update. The way `ComputeParamters`' semantics is defined, we should expect two different parameter updates to arrive separate from another that look like this:

```
ComputeParamaters { enable_columnation_lgalloc: Some(true), enable_lgalloc_eager_reclamation: None }
ComputeParamaters { enable_columnation_lgalloc: None, enable_lgalloc_eager_reclamation: Some(true) }
```

(The first switches on lgalloc, the second leaves lgalloc switched on and also switches on eager reclamation.)

To support this, we'd need to store the current lgalloc state in the `ComputeState`, to be able to refer back to it if it's not changed by the `ComputeParameters`.

Alternatively, this will resolve itself if we merge https://github.com/MaterializeInc/materialize/pull/25513, so you could also just add a TODO :)

Reply:
Yup, that's a good insight. I'll add a comment.